index.d.ts

```diff
@@ -0,0 +1,61 @@
+// Type definitions for gongfu
+// Project: https://github.com/hojberg/gongfu
+// Definitions by: Andrew Goodale <https://github.com/newyankeecodeshop>
+// TypeScript Version: 2.3
+
+import { Component } from "react";
+
+export interface Msg {
+  tag: string;
+}
+
+export type MsgConstructor = (a?: any) => Msg;
```

Comment:
`MsgConstructor` definitely takes any number of arguments of any type.

Reply:
It looks like it's only ever called with one argument:
https://github.com/hojberg/gongfu/blob/d71a61e60d7d013c36b664dba906317146dab23b/src/updater_for.ts#L6
Or maybe it's just `updaterFor` only supports one argument?

Reply:
@newyankeecodeshop yeah `updaterFor` expects one that only has 1 argument for the nested Msg. Perhaps we should have a specific type for that.